packages/block-editor/src/components/block-pattern-setup/index.js

```diff
@@ -178,10 +260,14 @@ const BlockPatternSetup = ( {
 					activeSlide={ activeSlide }
 					totalSlides={ patterns.length }
 					handleNext={ () => {
-						setActiveSlide( ( active ) => active + 1 );
+						setActiveSlide( ( active ) =>
+							Math.min( active + 1, patterns.length - 1 )
+						);
```

Comment:
Now that previous/next buttons are no longer disabled ([see `CarouselNavigation`](https://github.com/WordPress/gutenberg/pull/55425/files#r1362599890)), this is necessary to keep the active slide within bounds.